### PR TITLE
RAIN-1056: egov-indexer - Maven dependency changes and Merging to the Master branch

### PR DESCRIPTION
**Pull request by gopesh67**

### CODE REVIEW

File: egov-indexer/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.1.1 - 2020-06-04
+- Added typescript definition generation plugin
+- Upgraded to `tracer:2.0.0-SNAPSHOT`
+- Upgraded to spring boot `2.2.6-RELEASE`
+- Upgraded to flyway `6.4.3`
+- Increased version of this module to 0.1.0
```
Comment (tarunlalwani) [inline egov-indexer/CHANGELOG.md:11]: Remove this line, not needed

File: egov-indexer/pom.xml [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
     <groupId>org.springframework.boot</groupId>
     <artifactId>spring-boot-starter-parent</artifactId>
-    <version>1.5.22.RELEASE</version>
+    <version>2.2.6.RELEASE</version>
     <relativePath></relativePath>
   </parent>
   <groupId>org.egov</groupId>
   <artifactId>egov-indexer</artifactId>
-  <version>0.0.1-SNAPSHOT</version>
+  <version>0.1.0-SNAPSHOT</version>
```
Comment (tarunlalwani) [inline egov-indexer/pom.xml:13]: why is this `0.1.0`? Minimum version of any service in production has to be `1.0.0`

File: egov-indexer/pom.xml [hunk 3]
```diff
@@ -148,6 +149,36 @@
...
+            <class>org.egov.infra.indexer.web.contract.LegacyIndexRequest</class>
+            <class>org.egov.infra.indexer.web.contract.ReindexResponse</class>
+            <class>org.egov.infra.indexer.web.contract.LegacyIndexResponse</class>
+          </classes>
+          <customTypeMappings>
+            <customTypeMapping>org.egov.common.contract.request.RequestInfo:RequestInfo</customTypeMapping>
+            <customTypeMapping>org.egov.common.contract.response.ResponseInfo:ResponseInfo</customTypeMapping>
+          </customTypeMappings>
+          <namespace>Digit</namespace>
+          <debug>true</debug>
```
Comment (tarunlalwani) [inline egov-indexer/pom.xml:178]: Remove `debug`

File: egov-indexer/src/main/java/org/egov/infra/indexer/consumer/PGRCustomIndexMessageListener.java [hunk 4]
```diff
@@ -38,6 +38,7 @@ public void onMessage(ConsumerRecord<String, String> data) {
 		log.info("Topic: " + data.topic());
 		ObjectMapper mapper = indexerUtils.getObjectMapper();
 		try {
+			log.info("data.value()--->"+data.value());
```
Comment (tarunlalwani) [inline egov-indexer/src/main/java/org/egov/infra/indexer/consumer/PGRCustomIndexMessageListener.java:41]: Why is this there? It should not be there in a production build

File: egov-indexer/src/main/java/org/egov/infra/indexer/consumer/PTCustomIndexMessageListener.java [hunk 5]
```diff
@@ -38,7 +38,7 @@
 	 * index 5. Core indexing
 	 */
 	public void onMessage(ConsumerRecord<String, String> data) {
-		log.info("Topic: " + data.topic());
+		//log.info("Topic: " + data.topic());
```
Comment (tarunlalwani) [inline egov-indexer/src/main/java/org/egov/infra/indexer/consumer/PTCustomIndexMessageListener.java:41]: don't comment, remove the line

File: egov-indexer/src/main/java/org/egov/infra/indexer/service/DataTransformationService.java [hunk 6]
```diff
@@ -199,7 +210,13 @@ public DocumentContext enrichDataUsingExternalServices(DocumentContext documentC
 					String[] expressionArray = (fieldMapping.getOutJsonPath()).split("[.]");
 					String expression = indexerUtils.getProcessedJsonPath(fieldMapping.getOutJsonPath());
 					try {
-						Object value = JsonPath.read(mapper.writeValueAsString(response), fieldMapping.getInjsonpath());
+						String inputJsonPath = fieldMapping.getInjsonpath();
```
Comment (tarunlalwani) [inline egov-indexer/src/main/java/org/egov/infra/indexer/service/DataTransformationService.java:213]: Why is this poorly indented?

File: egov-indexer/src/main/java/org/egov/infra/indexer/producer/IndexerProducer.java [hunk 7]
```diff
@@ -12,7 +12,7 @@
 	public static final Logger logger = LoggerFactory.getLogger(IndexerProducer.class);
 
 	@Autowired
-    private CustomKafkaTemplate<String, Object> kafkaTemplate;
+    private KafkaTemplate<String, Object> kafkaTemplate;
```
Comment (tarunlalwani) [inline egov-indexer/src/main/java/org/egov/infra/indexer/producer/IndexerProducer.java:15]: Any reason to move to `KafkaTemplate`?

File: egov-indexer/src/main/java/org/egov/infra/indexer/service/LegacyIndexService.java [hunk 8]
```diff
@@ -159,22 +154,24 @@ public Boolean beginLegacyIndex(LegacyIndexRequest legacyIndexRequest) {
...
 					Integer presentCount = 0;
+					Integer maxRecords = legacyIndexRequest.getApiDetails().getPaginationDetails().getMaxRecords();
 					Integer size = null != legacyIndexRequest.getApiDetails().getPaginationDetails().getMaxPageSize()
 							? legacyIndexRequest.getApiDetails().getPaginationDetails().getMaxPageSize()
 							: defaultPageSizeForLegacyindex;
 					Boolean isProccessDone = false;
+
 					while (!isProccessDone) {
+						if (maxRecords > 0 && presentCount >= maxRecords ) {
+							log.info("Stopping since maxRecords have been processed: ", maxRecords);
```
Comment (tarunlalwani) [inline egov-indexer/src/main/java/org/egov/infra/indexer/service/LegacyIndexService.java:171]: Please correct the indentation of this file

File: egov-indexer/src/main/resources/application.properties [hunk 9]
```diff
@@ -88,11 +88,13 @@ egov.pt.host=http://pt-services-v2:8080
...
-egov.indexer.yml.repo.path=file://home/vishal/git/master/egov-services/core/egov-indexer/src/main/resources/rainmaker-pt-indexer.yml
+egov.indexer.yml.repo.path=file:///Users/nithin/Documents/eGov/egov-repos/core-services/egov-indexer/src/main/resources/collection-indexer.yml
 
 
 logging.pattern.console=%clr(%X{CORRELATION_ID:-}) %clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(${LOG_LEVEL_PATTERN:-%5p}) %clr(${PID:- }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}
+
+spring.main.allow-bean-definition-overriding: true
```
Comment (tarunlalwani) [inline egov-indexer/src/main/resources/application.properties:100]: Overriding bean should not be allowed, please fix the issue where bean is being overridden

File: egov-indexer/src/main/resources/application.properties [hunk 10]
```diff
@@ -8,20 +9,19 @@ egov.infra.indexer.host=https://egov-micro-dev.egovernments.org/elasticsearch
...
-flyway.baseline-on-migrate=true
-flyway.outOfOrder=true
-flyway.locations=db/migration/main
-flyway.enabled=true
+spring.flyway.url=jdbc:postgresql://localhost:5432/devdb
+spring.flyway.user=postgres
+spring.flyway.password=postgres
+spring.flyway.baseline-on-migrate=true
+spring.flyway.outOfOrder=true
+spring.flyway.locations=db/migration/main
```
Comment (tarunlalwani) [inline egov-indexer/src/main/resources/application.properties:23]: This needs to be `spring.flyway.locations=classpath:/db/migration/main`

File: egov-indexer/src/main/java/org/egov/infra/indexer/producer/IndexerProducer.java (hunk 7 above)
Comment (gopesh67) [reply egov-indexer/src/main/java/org/egov/infra/indexer/producer/IndexerProducer.java:15]: These changes were done by Abhilash for UAT. Looks like they wanted to use different version of kafka than tracer on old springboot  for using kafka connect.

File: egov-indexer/CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+
+
+# Changelog
+All notable changes to this module will be documented in this file.
+
+## 1.1.0 - 2020-06-04
+
+- Added typescript definition generation plugin
```
Comment (tarunlalwani) [inline egov-indexer/CHANGELOG.md:8]: add details about moving to Kafka connect as well

**Pull request merged**